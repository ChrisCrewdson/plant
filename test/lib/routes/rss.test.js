const bson = require('bson');

const { ObjectID } = bson;

/** @type {BizPlant[]} */
const plants = [{
  _id: 'pid2',
  locationId: 'lid1',
  userId: 'uid1',
  title: 'Test Title',
}];

const _id = new ObjectID('012345678901234567890123');
const userId = new ObjectID('012345678901234567890123');

const mockMongo = () => ({
  getNotesLatest: async () => {
    /** @type {DbNoteWithPlants[]} */
    const dbNoteWithPlants = [{
      _id,
      date: 20170606,
      plants,
      plantIds: [],
      userId,
    }];
    return Promise.resolve(dbNoteWithPlants);
  },
});

jest.mock('../../../lib/db/mongo', () => mockMongo);

const rss = require('../../../lib/routes/rss.js');

describe('rss', () => {
  test('renders xml', (done) => {
    const req = {
      protocol: 'fake protocol',
      get: () => 'fake host',
    };
    const res = {
      status: () => res,
      /** @param {string} builtXml */
      send: (builtXml) => {
        expect(builtXml).toMatchSnapshot();
        done();
      },
    };
    /** @type {import("express").Application} */
    const app = {
      // @ts-ignore - ignore for testing
      get: (_, routeAction) => {
        routeAction(req, res);
      },
    };
    rss(app);
  });
});
