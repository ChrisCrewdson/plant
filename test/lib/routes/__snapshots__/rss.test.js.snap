// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rss renders xml 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<rss version=\\"2.0\\" xmlns:atom=\\"http://www.w3.org/2005/Atom\\">
  <channel>
    <title>Latest Notes</title>
    <atom:link href=\\"fake protocol://fake host/rss\\" rel=\\"self\\" type=\\"application/rssxml\\" />
    <language>en-us</language>
    <link>fake protocol://fake host/rss</link>
    <description>Plant progress.</description>
    <item>
        <title>Test Title</title>
        <description><![CDATA[(no comment)]]></description>
        <pubDate>Tue, 06 Jun 2017 00:00:00 PST</pubDate>
        <guid>fake protocol://fake host/plant/test-title/undefined?noteid=012345678901234567890123#012345678901234567890123</guid>
        <link>fake protocol://fake host/plant/test-title/undefined?noteid=012345678901234567890123#012345678901234567890123</link>
      </item>
  </channel>
</rss>"
`;
